app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -74,18 +74,35 @@ class SupplementalApplicationPage extends React.Component {
   }
 
   handleSavePreference = async (preferenceIndex, application) => {
+    const { persistedApplication } = this.state
     const responses = await Promise.all([
       updateTotalHouseholdRent(application.id, application.total_monthly_rent),
       updatePreference(application.preferences[preferenceIndex])
     ])
+    const failed = some(responses, response => response === false)
+
+    if (!failed) {
+      const updateApplication = cloneDeep(persistedApplication)
+      updateApplication.preferences[preferenceIndex] = application.preferences[preferenceIndex]
+      this.setState({
+        persistedApplication: updateApplication,
+        confirmedPreferencesFailed: false
+      })
+    } else {
+      this.setState({ confirmedPreferencesFailed: true })
+    }
 
-    this.setState({
-      confirmedPreferencesFailed: some(responses, response => response === false)
-    })
+    return !failed
   }
 
-  handleOnDismissError = () => {
-    this.setState({ confirmedPreferencesFailed: false })
+  handleCancelPreferencePanel = (preferenceIndex) => {
```

Comment:
@fddayan Let's take a look at this function tomorrow in check-in - there may be some optimization we can do here.

Reply:
actually, this has some unnecessary code. I'm pushing a new version.